### PR TITLE
feat: chunk sorting

### PR DESCRIPTION
**Pull request by tustvold**
This should be the last prerequisite of automated chunk movement :grin:

### CODE REVIEW

File: server/src/db/catalog.rs [hunk 1]
```diff
@@ -119,6 +128,37 @@ impl Catalog {
...
+    /// Returns all chunks within the catalog in an arbitrary order
+    pub fn chunks(&self) -> Vec<Arc<RwLock<Chunk>>> {
+        let mut chunks = Vec::new();
+        let partitions = self.partitions.read();
+
+        for partition in partitions.values() {
+            let partition = partition.read();
+            chunks.extend(partition.chunks().cloned())
+        }
+        chunks
```
Comment (alamb) [inline server/src/db/catalog.rs:141]:
```suggestion
        let partitions = self.partitions.read();

        partitions.values()
          .flat_map(|partition| {
            let partition = partition.read();
            (partition.chunks().cloned()
          })
        .collect()
```

You could even avoid the `collect` if you wanted to change the signature to 

```
    pub fn chunks(&self) -> impl Iterator<Item=Arc<RwLock<Chunk>>> {
```

**Review by alamb — COMMENTED** [review]
This likely conflicts with https://github.com/influxdata/influxdb_iox/pull/1079 -- I think we should merge this PR first and then I'll rebase #1079

**Comment by alamb** [comment]
> @alamb Given this PR adds logic to the Catalog you'd rather wasn't - and #1079 doesn't - lets go with that one first and I can adapt and rebase this one

Done @tustvold

File: server/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ edition = "2018"
 [dependencies] # In alphabetical order
 arrow_deps = { path = "../arrow_deps" }
 async-trait = "0.1"
-bytes = "1.0"
+bytes = { version = "1.0", features = ["serde"] }
```
Comment (tustvold) [inline server/Cargo.toml:10]: This is necessary because of #1069

File: server/src/db/catalog.rs (hunk 1 above)
Comment (tustvold) [reply server/src/db/catalog.rs:134]: I don't think you can as the iterator returned by partition.chunks().cloned() needs to outlive the closure body, but it cannot outlive the lock guard which lives as long as the closure body

**Review by alamb — APPROVED** [review]
Reworked version looks 👌  Very nice. Thanks @tustvold

**Pull request merged**